### PR TITLE
4 Baystation12 PR's

### PR DESCRIPTION
**Pull request by tigercat2000**
This commit ports the following pull-requests:
 - Baystation12/Baystation12#8527
 - Baystation12/Baystation12#8921
 - Baystation12/Baystation12#8836

It also semi-ports (more details later)
 - Baystation12/Baystation12#8968

Features added:
 - Bartender can now change the barsign by hitting it with his ID. Anyone with access_bar can.
  - 8968 did not actually add this, it just added new sprites, which this does not, as we already have them. I only reference it because I do not know the original PR that added bar sign changes.
 - Clicking on the floor with tape will create a hazard strip, like in science. Clicking on the strip with the tape will remove it.
 - New comfy-chair colors; Red, Blue, Purple, Green.
 - Comfy Chairs now add an armrest overlay when mobs are buckled to them. 
 - Shower Curtains have two new versions; Security and Engineering. Currently unused, since we don't have showers there like baystation does, but worth adding for the future.

Refactors:
 - Beds now call a proc named afterbuckle when someone is buckled/unbuckled into them. By default, this does nothing.
  - Comfy chairs take advantage of this by using it to add/remove an overlay for the arm rest.
 - Comfy Chairs greyscaled and new arm rest overlay
  - Comfy Chairs are now colored by byond's color variable instead of premade sprites
  - Armrest shows up on top of the mob, as one would expect it too.
 - Curtains now use a #define for their layers. 
  - SHOWER_OPEN_LAYER is equal to OBJECT_LAYER + 0.4
  - SHOWER_CLOSED_LAYER is equal to MOB_LAYER + 0.1

Promotional image:
Example of most of these ports

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
yesyesyeyesyesyesye

**Comment by TheDZD** [comment]
:+1: 
We need more bar signs.

File: code/game/objects/structures/barsign.dm [hunk 1]
```diff
@@ -2,6 +2,7 @@
 	icon = 'icons/obj/barsigns.dmi'
 	icon_state = "empty"
 	anchored = 1
+	var/cult = 0
 	New()
 		ChangeSign(pick("magmasea", "limbo", "rustyaxe", "armokbar", "brokendrum", "meadbay", "thecavern", "cindikate", "theorchard", "lv426", "zocalo", "4theemprah", "ishimura", "tardis", "quarks", "tenforward", "thepranicngpony", "vault13", "solaris", "thehive", "cantina", "theouterspess", "milliways42", "thetimeofeve", "spaceasshole", "dwarffortress", "thebark", "thedrunkcarp", "theharmbaton", "thenest", "officerbeersky", "thesingulo"))
```
Comment (TheDZD) [inline code/game/objects/structures/barsign.dm]: Might want to add the new signs here too.

**Comment by tigercat2000** [comment]
@TheDZD We actually have all the barsigns that bay added in that PR, it was just the easiest thing to latch onto for the sign-changing code

**Comment by tigercat2000** [comment]
Played with git a bit, cult variable is now nopick

**Comment by ZomgPonies** [comment]
Yeah I cleaned up barsigns in general recently, removing the lower quality ones and adding in new animated ones.

A lot of them were just ugly, single sprite rehashes of really old tg/goon memes, the better more recent ones are animated, look unique from each other, and are server/lore agnostic.

**Pull request merged**